### PR TITLE
Remove all usage of `BaseDifficulty` (and access `Difficulty` instead)

### PR DESCRIPTION
**Pull request by peppy**
Preparation work for upcoming `BeatmapInfo` ctor improvements.

### CODE REVIEW

File: osu.Game/Beatmaps/BeatmapInfo.cs [hunk 1]
```diff
@@ -165,13 +165,6 @@ public int RulesetID
             }
         }
 
-        [Ignored]
-        public BeatmapDifficulty BaseDifficulty
-        {
-            get => Difficulty;
-            set => Difficulty = value;
-        }
```
Comment (bdach) [inline osu.Game/Beatmaps/BeatmapInfo.cs:173]: Do we care about external rulesets potentially still having references to this? As in would we want this to be around for a bit more, but just marked as obsolete, or is this a hard blocker for further work?

Comment (peppy) [reply osu.Game/Beatmaps/BeatmapInfo.cs:173]: do you have an example of this being used by a rule set?

Comment (smoogipoo) [reply osu.Game/Beatmaps/BeatmapInfo.cs:173]: One example is similar to the usage which was changed here: https://github.com/taulazer/tau/blob/master/osu.Game.Rulesets.Tau/Beatmaps/TauBeatmapConverter.cs

Comment (peppy) [reply osu.Game/Beatmaps/BeatmapInfo.cs:173]: Fair. Will add back as `Obsolete` for this case.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**